### PR TITLE
Fix/pr 466 max arias full width container

### LINKED ISSUE #406: Carousel has no width if parent uses Flexbox
[Issuehunt badges](https://issuehunt.io/r/brainhubeu/react-carousel/issues/406)

**Describe the bug**
I am using the carousel inside a section element which uses Flexbox, but the `BrainhubCarousel__track` and the `BrainhubCarouselItem` elements are all getting `width: 0px; max-width: 0px; min-width: 0px;`

**To Reproduce**
I created this [CodeSandbox](https://codesandbox.io/s/reverent-clarke-pljuc) which reproduces the issue.

**Expected behavior**
Slides and track elements should get the appropriate width.

**Screenshots**

**Environment**
```
System:
    OS: macOS 10.15.3
    CPU: (8) x64 Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz
    Memory: 241.82 MB / 16.00 GB
  Binaries:
    Node: 10.13.0 - ~/.nvm/versions/node/v10.13.0/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v10.13.0/bin/npm
  Browsers:
    Chrome: 80.0.3987.149
    Firefox: 74.0
    Safari: 13.0.5
```

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $56.00)

- [piotr-s-brainhub piotr-s-brainhub](https://issuehunt.io/u/piotr-s-brainhub) ($56.00)

### Submitted pull Requests
- [#466 Adds width to the carousel container, adds example in docs](https://issuehunt.io/r/brainhubeu/react-carousel/pull/466)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/brainhubeu/react-carousel/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by piotr-s-brainhub**
Deployed to https://beghp.github.io/gh-pages-rc-1<br>It fixes #406 <br>- [x] an issue linked to the PR<br><br><br><br>

### CODE REVIEW

**Comment by piotr-s-brainhub** [comment]
Thanks @max-arias 
I checked it's fixed and nothing is broken (desktop & mobile).

File: src/styles/Carousel.scss [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
+.BrainhubCarousel__container {
+  width: 100%;
 
-  &.BrainhubCarousel--isRTL {
-    direction: rtl;
-
-    .BrainhubCarousel__trackContainer {
-      .BrainhubCarousel__track {
-        direction: rtl;
+  .BrainhubCarousel {
```
Comment (humbak) [inline src/styles/Carousel.scss:4]: do we really need nesting everything and adding more specificity to rest selectors?

File: src/styles/Carousel.scss [hunk 2]
```diff
@@ -1,29 +1,33 @@
-.BrainhubCarousel {
-  overflow: hidden;
-  display: flex;
-  align-items: center;
+.BrainhubCarousel__container {
+  width: 100%;
```
Comment (humbak) [inline src/styles/Carousel.scss:2]: this solution will not work well with `flexFlow: row`

File: docs/examples/flexContainer.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+## Flex Container
+
+You can use a Carousel within a flex container
+```jsx render
+
+<div style={{ display: 'flex', justifyContent: 'center', flexFlow: 'column', alignContent: 'center', textAlign: 'center' }}>
```
Comment (humbak) [inline docs/examples/flexContainer.md:6]: what about more popular `flexDirection: 'row'` ?

File: docs-www/package.json [hunk 4]
```diff
@@ -15,7 +15,7 @@
   "dependencies": {
     "@brainhubeu/gatsby-docs-kit": "3.0.8",
     "@brainhubeu/react-carousel": "^1.13.44",
-    "gatsby": "2.20.10",
+    "gatsby": "2.20.24",
```
Comment (humbak) [inline docs-www/package.json:18]: Should we update `gatsby` in this MR? Is this necessary?

File: src/styles/Carousel.scss (hunk 1 above)
Comment (max-arias) [reply src/styles/Carousel.scss:4]: Sure, we don't need it, but I don't think it hurts to add a little specificity.

File: docs-www/package.json (hunk 4 above)
Comment (max-arias) [reply docs-www/package.json:18]: I could not get `docs-www` to work with the Gatsby version we had, I bumped it to allow running it

File: src/styles/Carousel.scss (hunk 2 above)
Comment (max-arias) [reply src/styles/Carousel.scss:2]: I'll look into this

Comment (max-arias) [reply src/styles/Carousel.scss:2]: @humbak This seems to work fine with flex-flow row, see: https://imgur.com/a/dEFQE7k

Comment (humbak) [reply src/styles/Carousel.scss:2]: @max-arias use more text and you will see something like it: 
image

File: src/styles/Carousel.scss (hunk 1 above)
Comment (humbak) [reply src/styles/Carousel.scss:4]: Hurts or not huts that is just bad practice. We need to correct it before merging. But the more important thing is about `flex-direction: row`.

File: src/styles/Carousel.scss (hunk 2 above)
Comment (max-arias) [reply src/styles/Carousel.scss:2]: Thanks, updated. Let me know.

File: src/styles/Carousel.scss (hunk 1 above)
Comment (max-arias) [reply src/styles/Carousel.scss]: Fair enough, should be fixed now

**Comment by max-arias** [comment]
FYI issues discussed are fixed, but we need to update this branch with master

**Comment by piotr-s-brainhub** [comment]
@max-arias 

Thanks for resolving comments but could you also address this comment https://github.com/brainhubeu/react-carousel/pull/467#discussion_r422203939 ?

I just resolved the conflicts.

**Comment by max-arias** [comment]
@piotr-s-brainhub Yup, that should be fixed 👍

**Comment by piotr-s-brainhub** [comment]
@humbak 

could you re-review?

**Comment by humbak** [comment]
I think we have it!

**Comment by piotr-s-brainhub** [comment]
@max-arias 

I checked it fixes the problem and it doesn't break anything (desktop & mobile).

On mobile, in the flex example, the images are shrunk horizontally which doesn't look good but it's related to another issue and It works better than previously (I checked that on mobile, as well, flex makes a blank space in production docs).

**Pull request merged**